tools/kata-agent-ctl/src/main.rs

```diff
@@ -0,0 +1,296 @@
+
+const ABOUT_TEXT: &str = "Kata Containers agent tool";
+
+const WARNING_TEXT: &str = r#"WARNING:
+    This tool is for *advanced* users familiar with the low-level agent API calls.
+    Further, it is designed to be run on test and development systems **only**:
+    since the tool can make arbitrary API calls, it is possible to easily confuse
+    irrevocably other parts of the system or even kill a running container or
+    sandbox."#;
+
+fn make_examples_text(program_name: &str) -> String {
+    let bundle = "$bundle_dir";
+    let cid = 3;
+    let container_id = "$container_id";
+    let config_file_uri = "file:///tmp/config.json";
+    let port = 1024;
+    let sandbox_id = "$sandbox_id";
+
+    format!(
+        r#"EXAMPLES:
+
+- Check if the agent is running:
+
+  $ {program} connect --vsock-cid {cid} --vsock-port {port} --cmd Check
+
+- Query the agent environment:
+
+  $ {program} connect --vsock-cid {cid} --vsock-port {port} --cmd GuestDetails
+
+- List all available (built-in and Kata Agent API) commands:
+
+  $ {program} connect --vsock-cid {cid} --vsock-port {port} --cmd list
+
+- Generate a random container ID:
+
+  $ {program} generate-cid
+
+- Generate a random sandbox ID:
+
+  $ {program} generate-sid
+
+- Attempt to create 7 sandboxes, ignoring any errors:
+
+  $ {program} connect --vsock-cid {cid} --vsock-port {port} --repeat 7 --cmd CreateSandbox
+
+- Query guest details forever:
+
+  $ {program} connect --vsock-cid {cid} --vsock-port {port} --repeat -1 --cmd GuestDetails
+
+- Send a 'SIGUSR1' signal to a container process:
+
+  $ {program} connect --vsock-cid {cid} --vsock-port {port} --cmd 'SignalProcess signal=usr1 sid={sandbox_id} cid={container_id}'
+
+- Create a sandbox with a single container, and then destroy everything:
+
+  $ {program} connect --vsock-cid {cid} --vsock-port {port} --cmd CreateSandbox
+  $ {program} connect --vsock-cid {cid} --vsock-port {port} --bundle-dir {bundle:?} --cmd CreateContainer
+  $ {program} connect --vsock-cid {cid} --vsock-port {port} --cmd DestroySandbox
+
+- Create a Container using a custom configuration file:
+
+  $ {program} connect --vsock-cid {cid} --vsock-port {port} --bundle-dir {bundle:?} --cmd 'CreateContainer spec={config_file_uri}'
+	"#,
+        bundle = bundle,
+        cid = cid,
+        config_file_uri = config_file_uri,
+        container_id = container_id,
+        port = port,
+        program = program_name,
+        sandbox_id = sandbox_id,
+    )
+}
+
+fn connect(name: &str, global_args: clap::ArgMatches) -> Result<()> {
+    let args = match global_args.subcommand_matches("connect") {
+        Some(args) => args,
+        None => return Err("BUG: missing sub-command arguments".to_string()),
+    };
+
+    let interactive = args.is_present("interactive");
+    let ignore_errors = args.is_present("ignore-errors");
+
+    let cid_str = match args.value_of("vsock-cid") {
+        Some(s) => s,
+        None => return Err("need VSOCK cid".to_string()),
+    };
```

Comment:
Here again, I would use `map_err` instead of `match`.

Reply:
Done.